src/hooks/usePools.ts

```diff
@@ -0,0 +1,163 @@
+import { POOL_DEPLOYER_ADDRESS } from '../constants/v3/addresses';
+import { Currency, Token } from '@uniswap/sdk-core';
+import { useMemo } from 'react';
+import { useActiveWeb3React } from 'hooks';
+import abi from 'constants/abis/v3/pool.json';
+import { Interface } from '@ethersproject/abi';
+import { useInternet } from './useInternet';
+import { useToken } from './Tokens';
+import { usePreviousNonErroredArray } from './usePrevious';
+import { Pool } from 'lib/src/pool';
+import { computePoolAddress } from './v3/computePoolAddress';
+import { useMultipleContractSingleData } from 'state/multicall/v3/hooks';
+
+const POOL_STATE_INTERFACE = new Interface(abi);
+
+export enum PoolState {
+  LOADING,
+  NOT_EXISTS,
+  EXISTS,
+  INVALID,
+}
+
+export function usePools(
+  poolKeys: [Currency | undefined, Currency | undefined][],
+): [PoolState, Pool | null][] {
+  const { chainId } = useActiveWeb3React();
+
+  const transformed: ([Token, Token] | null)[] = useMemo(() => {
+    return poolKeys.map(([currencyA, currencyB]) => {
+      if (!chainId || !currencyA || !currencyB) return null;
+
+      const tokenA = currencyA?.wrapped;
+      const tokenB = currencyB?.wrapped;
+      if (!tokenA || !tokenB || tokenA.equals(tokenB)) return null;
+      const [token0, token1] = tokenA.sortsBefore(tokenB)
+        ? [tokenA, tokenB]
+        : [tokenB, tokenA];
+      return [token0, token1];
+    });
+  }, [chainId, poolKeys]);
+
+  const poolAddresses: (string | undefined)[] = useMemo(() => {
+    const poolDeployerAddress = chainId && POOL_DEPLOYER_ADDRESS[chainId];
+
+    return transformed.map((value) => {
+      if (!poolDeployerAddress || !value) return undefined;
+
+      return computePoolAddress({
+        poolDeployer: poolDeployerAddress,
+        tokenA: value[0],
+        tokenB: value[1],
+      });
+    });
+  }, [chainId, transformed]);
+
+  const globalState0s = useMultipleContractSingleData(
+    poolAddresses,
+    POOL_STATE_INTERFACE,
+    'globalState',
+  );
+
+  // TODO: This is a bug, if all of the pool addresses error out, and the last call to use pools was from a different hook
+  // You will get the results which don't match the pool keys
+  const prevGlobalState0s = usePreviousNonErroredArray(globalState0s);
```

Comment:
this should be investigated in a bit more.  I don't think it can be an issue currently since the pool keys wont match.